### ISSUE TITLE
Cannot use hash in spec: `More than one spec in string`

### ISSUE THREAD

**Issue by eschnett** [spack/spack#1267]
I have two packages `cactusext` installed:

```
$ spack find -l cactusext
==> 2 installed packages.
-- darwin-elcapitan-x86_64 / gcc@6.1.0-spack --------------------
vyakp7b cactusext@master  lpdwn2k cactusext@master
```

They are very similar (some details in their dependencies differ), and I need to distinguish between them via their spec. So I write:

```
spack view -d true symlink ~/spack/cactusdeps cactusext/lpdwn2k
```

but this does not work:

```
Traceback (most recent call last):
  File "/Users/eschnett/src/spack/bin/spack", line 179, in <module>
    main()
  File "/Users/eschnett/src/spack/bin/spack", line 157, in main
    return_val = command(parser, args)
  File "/Users/eschnett/src/spack/lib/spack/spack/cmd/view.py", line 286, in view
    seeds = [spack.cmd.disambiguate_spec(s) for s in args.specs]
  File "/Users/eschnett/src/spack/lib/spack/spack/cmd/__init__.py", line 136, in disambiguate_spec
    matching_specs = spack.installed_db.query(spec)
  File "/Users/eschnett/src/spack/lib/spack/spack/database.py", line 574, in query
    if query_spec is any or rec.spec.satisfies(query_spec):
  File "/Users/eschnett/src/spack/lib/spack/spack/spec.py", line 1712, in satisfies
    other = self._autospec(other)
  File "/Users/eschnett/src/spack/lib/spack/spack/spec.py", line 1691, in _autospec
    spec = spack.spec.Spec(spec_like)
  File "/Users/eschnett/src/spack/lib/spack/spack/spec.py", line 489, in __init__
    raise ValueError("More than one spec in string: " + spec_like)
ValueError: More than one spec in string: cactusext/lpdwn2k
```

I tried using more or fewer characters in the hash code, but that did not make a difference.

**Comment by adamjstewart** [comment]
@eschnett The problem is that `/lpdwn2k` already includes the name of the package, so `cactusext/lpdwn2k` expands to `cactusext cactusext@1.2.3+foo ^dependency@3.2.1+bar ...`. As the error message states, you cannot specify more than one spec at a time. The solution is to only specify the name of the package once:
```
$ spack view -d true symlink ~/spack/cactusdeps /lpdwn2k
```
P.S. I believe this limitation was fixed by @becker33 in #2889. Can you confirm?

**Comment by becker33** [comment]
@adamjstewart @eschnett I can confirm that #2889 should fix this, obviously if this is occurring with #2889 let me know so I can fix it.
